lib/postgrex/extensions/binary.ex

```diff
@@ -304,6 +306,31 @@ defmodule Postgrex.Extensions.Binary do
     [flags|bin]
   end
 
+  defp encode_hstore(bin) do
+    keys_and_values = Enum.reduce bin, "", fn ({key, value}, acc) ->
+        acc <> encode_hstore_key(key) <> encode_hstore_value(value)
+    end
+    <<Map.size(bin)::int32>> <> keys_and_values
+  end
+
+  defp encode_hstore_key(key) when is_nil(key) do
+    raise ArgumentError, message: "Hstore keys cannot be nil!"
+  end
+
+  defp encode_hstore_key(key) do
+    encode_hstore_value key
+  end
+
+  defp encode_hstore_value(nil) do
+    <<-1::int32>>
+  end
+
+  defp encode_hstore_value(value) do
+    string_value = to_string(value)
```

Comment:
ok I'll remove that. It's a pain in the ass, but you're right about consistency with the spec.